ckanext/canada/plugins.py

```diff
@@ -102,6 +103,17 @@ def get_validators(self):
                 'canada_security_upload_presence':
                     validators.canada_security_upload_presence}
 
+    # IAuthenticator
+    def identify(self):
+        if p.toolkit.request.path == p.toolkit.url_for(u'user.login'):
```

Comment:
@RabiaSajjad you will have to move your `abort` with the message from the helpers into this method. and just use the `if not helpers.registry_network_access()` condition.

`request` should also already be imported here, so you can change `p.toolkit.request` to just `request`.

And the `path == p.toolkit.url_for(u'user.login')` will not work in all cases as it will not pass for the `/en` and `/fr` paths.

The best thing would be to do it by the Flask Blueprints so that if any routes change, this will pick it up. Something like this:
```
        controller, action = p.toolkit.get_endpoint()
        blueprint = '.'.join((controller, action))
        restricted_blueprints = [
            'canada.login',
            'user.login',
        ]
        if blueprint in restricted_blueprints....
```

This would also allow us to restrict the API blueprints like `canada.action` / `api.action`